### ISSUE TITLE
djxl fails to decode 32-bit float JXL to PNG

### ISSUE THREAD

**Issue by Traneptora** [libjxl/libjxl#3018]
**Describe the bug**
Take a 32-bit float JXL, such as the one that is attached. It fails to decode to a PNG on the CLI.
[float.jxl.gz](https://github.com/libjxl/libjxl/files/13652586/float.jxl.gz)

**To Reproduce**
```
$ djxl float.jxl test.png
JPEG XL decoder v0.10.0 e986e5c9 [AVX2]
./lib/extras/enc/encode.cc:126: encoding 32-bit original to 16 bits
./lib/jxl/decode.cc:2823: Invalid bit depth 32 for uint16 output
JxlDecoderSetImageOutBitDepth failed
DecompressJxlToPackedPixelFile failed
```

**Expected behavior**
PNG doesn't support floating point, but this is an XYB-encoded VarDCT image so the tagged depth is a suggestion more than anything else. djxl should request 16-bit pixel data from libjxl rather than requesting 32-bit and then giving up when it can't send it to the PNG encoder.
```
$ jxlinfo -v float.jxl
JPEG XL image, 8x8, lossy, 32-bit float (8 exponent bits) RGB
num_color_channels: 3
num_extra_channels: 0
have_preview: 0
have_animation: 0
Intrinsic dimensions: 8x8
Orientation: 1 (Normal)
Color space: RGB, D65, sRGB primaries, sRGB transfer function, rendering intent: Perceptual
$ jxlatte --verbose float.jxl
Image: float.jxl
    Level: 5
    Size: 8x8
    Pixel Format: RGB
    Bit Depth: 32
    Extra Channels: 0
    XYB Encoded: true
    Primaries: sRGB / BT.709
    White Point: D65
    Transfer Function: sRGB / ISO-IEC 61966-2-1
    Lossless: No
Frame Info:
    Encoding: VarDCT
    Type: Regular
    Size: 8x8
    Origin: (0, 0)
    YCbCr: false
Decoded to pixels, discarding output.
```

**Environment**
 - OS: Arch Linux x86_64
 - Compiler version: GCC 13.2.1
 - cjxl/djxl version string: JPEG XL decoder v0.10.0 e986e5c9 [AVX2]

**Additional context**
I didn't attach `float.jxl` using the `.jpg` trick because it's an 8x8 white rectangle and I didn't want it to embed inline and get lost.

**Comment by eustas** [comment]
Going to take a look soon.

**Comment by mo271** [comment]
should be fixed by #3057
